### ISSUE TITLE
internal error: zero transaction timestamp in EvalContext

### ISSUE THREAD

**Issue by Cosmicoppai** [cockroachdb/cockroach#107857]
## Table
```sql
CREATE TABLE IF NOT EXISTS players (
    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
    name VARCHAR NOT NULL,
    country VARCHAR NOT NULL,
    dob DATE CHECK (dob <= CURRENT_DATE ),
    sex CHAR(1) CHECK (sex IN ('M', 'F')),
    height INT NOT NULL, /* height in cm */
    sport_id uuid REFERENCES sports(id),
    INDEX idx_player_name (name)
);

```

```python
async def load_players(file: str):
    await DataBase.connect()

    _data = []

    with open(file, "r") as player_file:
        csv_data = csv.DictReader(player_file)
        sport_ids = await fetch_sport_id(DataBase.connection, [row["Sport"] for row in csv_data])
        player_file.seek(0)
        csv_data.__init__(player_file)
        for row in csv_data:
            _data.append({"name": row["Name"], "country": row["Country"],
                          "dob": gen_date(), "sex": row["Sex"], "height": int(float(row["Height"])),
                          "sport_id": str(sport_ids[row["Sport"]])})

        data = [tuple(_data[i].values()) for i in range(len(_data))]

        try:
            logging.info("Executing Query")
            for idx in range(10000, len(data), 10000):
                await DataBase.connection.copy_records_to_table("players", records=data[:idx], columns=["name", "country", "dob", "sex", "height", "sport_id"],
                                                                schema_name="olympus")
                logging.info(f"Inserted {idx} records")
        except Exception as e:
            print(e)
            raise e

        finally:
            await DataBase.connection.close()
```

```error
Traceback (most recent call last):
  File "I:\olympus\backend\load_data.py", line 101, in <module>
    asyncio.run(load_players("data/stats/stats_data.csv"))
  File "C:\Users\KANAK\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Users\KANAK\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 641, in run_until_complete
    return future.result()
  File "I:\olympus\backend\load_data.py", line 68, in load_players
    raise e
  File "I:\olympus\backend\load_data.py", line 63, in load_players
    await DataBase.connection.copy_records_to_table("players", records=data[:idx], columns=["name", "country", "dob", "sex", "height", "sport_id"],
  File "I:\olympus\olympus_env\lib\site-packages\asyncpg\connection.py", line 983, in copy_records_to_table
    return await self._protocol.copy_in(
  File "asyncpg\protocol\protocol.pyx", line 529, in copy_in
asyncpg.exceptions.InternalServerError: internal error: zero transaction timestamp in EvalContext
DETAIL:  stack trace:
github.com/cockroachdb/cockroach/pkg/sql/sem/eval/context.go:626: GetTxnTimestamp()
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/builtins.go:8363: currentDate()
github.com/cockroachdb/cockroach/pkg/sql/sem/eval/expr.go:479: EvalFuncExpr()
github.com/cockroachdb/cockroach/bazel-out/k8-opt/bin/pkg/sql/sem/tree/eval_expr_generated.go:276: Eval()
github.com/cockroachdb/cockroach/pkg/sql/sem/eval/expr.go:26: Expr()
github.com/cockroachdb/cockroach/pkg/sql/opt/norm/fold_constants_funcs.go:694: FoldFunction()
github.com/cockroachdb/cockroach/bazel-out/k8-opt/bin/pkg/sql/opt/norm/factory.og.go:20488: ConstructFunction()
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scalar.go:571: buildFunction()
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scalar.go:334: buildScalar()
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scalar.go:322: buildScalar()
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/mutation_builder.go:798: addCheckConstraintCols()
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/insert.go:678: buildInsert()
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/insert.go:293: buildInsert()
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:344: func2()
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/with.go:116: processWiths()
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:343: buildStmt()
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:272: buildStmtAtRoot()
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:246: Build()
github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:574: buildExecMemo()
github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:238: makeOptimizerPlan()
github.com/cockroachdb/cockroach/pkg/sql/copy_from.go:1112: insertRowsInternal()
github.com/cockroachdb/cockroach/pkg/sql/copy_from.go:1038: insertRows()
github.com/cockroachdb/cockroach/pkg/sql/copy_from.go:669: processCopyData()
github.com/cockroachdb/cockroach/pkg/sql/copy_from.go:580: run()
github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:3149: func9()
github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:2986: execWithProfiling()
github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:3141: execCopyIn()
github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2356: execCmd()
github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2097: run()
github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:893: ServeConn()
github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:778: func1()
GOROOT/src/runtime/asm_amd64.s:1594: goexit()

HINT:  You have encountered an unexpected error.

Please check the public issue tracker to check whether this problem is
already tracked. If you cannot find it there, please report the error
with details by creating a new issue.

If you would rather not post publicly, please contact us directly
using the support form.

We appreciate your feedback.
```

**Environment:**

 - asyncpg

Jira issue: CRDB-30223

**Comment by yuzefovich** [comment]
It's likely a dup of #104456 fixed in 23.1.5. @Cosmicoppai what version were you using when you ran into this?

**Comment by Cosmicoppai** [comment]
> It's likely a dup of #104456 fixed in 23.1.5. @Cosmicoppai what version were you using when you ran into this?

I'm using a serverless cluster (ver: 23.1.3)

**Comment by yuzefovich** [comment]
I see, ok, then I'm closing this as a dup of #104456, and at some point serverless host clusters will be upgraded to 23.1.x that includes the fix.

Temporary workaround is to drop the CHECK constraint on `dob` column before performing the COPY, then do the COPY, and then add the constraint back.
